packages/pyright-internal/src/analyzer/program.ts

```diff
@@ -1890,12 +1890,16 @@ export class Program {
                 return false;
             }
 
-            if (!this._disableChecker) {
+            if (!this._disableChecker && !fileToCheck.sourceFile.isFileDeleted()) {
```

Comment:
I wonder if it would be better to add the check for `isFileDeleted()` within `_bindFile`? I guess it depends on whether there are other places that call `_bindFile` that do not already check for this condition.

Reply:
`_bindFile` actually already checks for this and skips binding if the file is deleted. I was thinking that maybe _bindFile should return a boolean indicating if it actually bound the file or not (instead of doing this check here), so callers could skip calling things that depend upon the bind happening.